src/MsalAuthProvider.ts

```diff
@@ -65,6 +66,8 @@ export abstract class MsalAuthProvider implements IAuthProvider {
         validateAuthority: authProviderConfig.validateAuthority,
       },
     );
+
+    this.checkIfUserAuthenticated();
```

Comment:
The MsalAuthProvider should be responsible for knowing the authentication state, and should inform any classes that extend from it.